src/util/triggers.ts

```diff
@@ -79,6 +79,38 @@ async function checkTrigger(
   now: Date
 ): Promise<PushTriggerUpdate> {
   switch (trigger.type) {
+    case 'all': {
+      const { triggers } = trigger
+      let done = true
+      let states = Array.isArray(state) ? state : []
+
+      for (let i = 0; i < triggers.length; ++i) {
+        const update = await checkTrigger(tools, triggers[i], states[i], now)
+        if (!update.done) done = false
+        if (update.state !== states[i]) {
+          states = [...states]
+          states[i] = update.state
+        }
+      }
+      return { done, state: states }
+    }
```

Comment:
I see an opportunity to short-circuit the loop with an early return if `!update.done`.

Reply:
Nope, I think we actually do want to check all the conditions. Maybe the second transaction is confirmed, but not the first one. We still want to check that and record the date.